egs2/chime7_task1/asr1/local/gen_task1_data.py

```diff
@@ -0,0 +1,517 @@
+            parents=True, exist_ok=True
+        )
+        Path(os.path.join(out_dir, "transcriptions_scoring", split)).mkdir(
+            parents=True, exist_ok=True
+        )
+        Path(os.path.join(out_dir, "uem", split)).mkdir(parents=True, exist_ok=True)
+
+    for split in splits:
+        json_dir = os.path.join(root_dir, "transcriptions", split)
+        ann_json = glob.glob(os.path.join(json_dir, "*.json"))
+        assert len(ann_json) > 0, (
+            "CHiME-6 JSON annotation was not found in {}, please check if "
+            "CHiME-6 data was downloaded correctly and the CHiME-6 main dir "
+            "path is set correctly".format(json_dir)
+        )
+        # we also create audio files symlinks here
+        audio_files = glob.glob(os.path.join(root_dir, "audio", split, "*.wav"))
+        sess2audio = {}
+        for x in audio_files:
+            session_name = Path(x).stem.split("_")[0]
+            if Path(x).stem.split("_")[-1].startswith("P") and eval_opt > 0:
+                continue
+            if session_name not in sess2audio:
+                sess2audio[session_name] = [x]
+            else:
+                sess2audio[session_name].append(x)
+
+        # for each json file
+        all_uem = []
+        for j_file in ann_json:
+            with open(j_file, "r") as f:
+                annotation = json.load(f)
+            sess_name = Path(j_file).stem
+
+            annotation, scoring_annotation = normalize_chime6(
+                annotation, scoring_txt_normalization, eval_opt
+            )
+
+            tsplit = None  # find chime7 destination split
+            for k in ["train", "dev", "eval"]:
+                if sess_name in chime6_map[k]:
+                    tsplit = k
+
+            # create symlinks too
+            if eval_opt == 0 and tsplit == "eval":
+                continue
+            if eval_opt > 0 and tsplit != "eval":
+                continue
+
+            [
+                os.symlink(
+                    x,
+                    os.path.join(out_dir, "audio", tsplit, Path(x).stem) + ".wav",
+                )
+                for x in sess2audio[sess_name]
+            ]
+
+            with open(
+                os.path.join(out_dir, "transcriptions", tsplit, sess_name + ".json"),
+                "w",
+            ) as f:
+                json.dump(annotation, f, indent=4)
+            # retain original annotation but dump also the scoring one
+            with open(
+                os.path.join(
+                    out_dir,
+                    "transcriptions_scoring",
+                    tsplit,
+                    sess_name + ".json",
+                ),
+                "w",
+            ) as f:
+                json.dump(scoring_annotation, f, indent=4)
+
+            first = sorted([float(x["start_time"]) for x in annotation])[0]
+            end = max([sf.SoundFile(x).frames for x in sess2audio[sess_name]])
```

Comment:
Why is here the `max` and not `min` operation used?
This is often annoying, since the CHiME-6 data has different length for each array. But usually they are just off by a few seconds or even less.

Minor: Using seconds as unit in the uem file and format, which uses `round` and not `floor`, sometimes causes errors, that the number that a user reads is larger than the true length.
Here, an example for S03 from CHiME-6
```
>>> float(f'{126140142 / 16000:.3f}') * 16000
126140144.0
```
A fix is relative long (see https://stackoverflow.com/a/51981641/5766934 or using [decimal.Decimal](https://docs.python.org/3/library/decimal.html)). Not sure if it should be done here. (Sadly, it is common to use seconds as units, which introduces always rounding issues.)

Alternatively, you could use `{:.6f}`, which disables the rounding error for a sample rate of 16000.

Reply:
> Why is here the max and not min operation used?

I think because it is an `UEM` max is "more correct" here. 
My reasoning was that we have to exclude the enrolling portion (which was not excluded in the last challenge) but all the other data is valid. If, for whatever reason e.g. mic choice or system design, a participant makes a prediction whose boundary falls into the longest recording (it will be an insertion likely) we have to consider it, as it is done on some audio data which is actually legit. 


> Using seconds as unit in the uem file and format

I agree, I have followed the UEM definition here by NIST https://web.archive.org/web/20170119114252/http://www.itl.nist.gov/iad/mig/tests/rt/2009/docs/rt09-meeting-eval-plan-v2.pdf 
Indeed it is prone to truncation errors, I am hesitant of changing it to samples because everyone (at least to my knowledge) uses this definition for UEMs (e.g. DIHARD challenges). 
`{:.6f}` could be a choice that could comply with NIST definition since they do not specify how many decimal digits it should have. They specify it has to be in seconds though. 
On the other hand I also think that rounding issues at 10ms scale would hardly impact the score at the end.

Reply:
Personally I think we should stick with the conventional 3 point decimal representation. Rounding errors should be dealt with by the system, since different systems would have different configuration. For example, in the GSS implementation using Lhotse, we set an "audio mismatch tolerance" which basically allows the mismatched channels to be padded with zeros. Other systems may have different ways of dealing with this problem.